pkg/cvo/sync_worker.go

```diff
@@ -971,21 +971,21 @@ func ownerRefModifier(config *configv1.ClusterVersion) resourcebuilder.MetaV1Obj
 
 // runThrottledStatusNotifier invokes fn every time ch is updated, but no more often than once
 // every interval. If bucket is non-zero then the channel is throttled like a rate limiter bucket.
-func runThrottledStatusNotifier(stopCh <-chan struct{}, interval time.Duration, bucket int, ch <-chan SyncWorkerStatus, fn func()) {
+func runThrottledStatusNotifier(ctx context.Context, interval time.Duration, bucket int, ch <-chan SyncWorkerStatus, fn func()) {
 	// notify the status change function fairly infrequently to avoid updating
 	// the caller status more frequently than is needed
 	throttle := rate.NewLimiter(rate.Every(interval), bucket)
-	wait.Until(func() {
-		ctx := context.Background()
+	wait.UntilWithContext(ctx, func(context.Context) {
+		localCtx := context.Background()
 		var last SyncWorkerStatus
 		for {
 			select {
-			case <-stopCh:
+			case <-ctx.Done():
 				return
 			case next := <-ch:
 				// only throttle if we aren't on an edge
 				if next.Generation == last.Generation && next.Actual == last.Actual && next.Reconciling == last.Reconciling && (next.Failure != nil) == (last.Failure != nil) {
-					if err := throttle.Wait(ctx); err != nil {
+					if err := throttle.Wait(localCtx); err != nil {
```

Comment:
Why do we need `localCtx` here?  Can we pass `ctx` through and... I guess catch and ignore [the context values][1] with:

```go
if err := throttle.Wait(ctx); err != nil && err != context.Canceled && err != context.DeadlineExceeded {
  utilruntime.HandleError(fmt.Errorf("unable to throttle status notification: %v", err))
}
```

[1]: https://golang.org/pkg/context/#pkg-variables

Reply:
My thinking was to make as little change as possible to the existing code and context handling and therefore maintained the local declaration of the local context there and just renamed it. Keeping it separate from the context now passed into the func. Couldn't find a "best-practice" on topic of when/if one should create a separate context but it does seem more clear to me to keep the two separate contexts.

Reply:
Removed localCtx in commit [8ee0d52](https://github.com/openshift/cluster-version-operator/pull/410/commits/8ee0d52e67593922f965e8c48bc22275acc62bab)